File: python/lib/terrain.py
class Terrain(object):
  def __init__(self, terrain_str):
    lat, long = terrain_str.split(' ')
    self.longitude = int(long) # int representing y axis length
    self.latitude = int(lat) # int representing x axis length
    self.reset_landscape()
            
    pass

  def reset_landscape(self):
    self.landscape = {}
    for i in xrange(0, self.longitude + 1):
      self.landscape[i] = [' '] * (self.latitude + 1)

File: elixir/lib/rover_challenge/terrain.ex
defmodule RoverChallenge.Terrain do
  defstruct [:lat, :long]

  def from(terrain_str) do
    [lat, long] = String.split(terrain_str, " ")
                  |> Enum.map(fn(e)-> String.to_integer(e) end)

    %__MODULE__{lat: lat, long: long}
  end

  def landscape(terrain) do
    Enum.reduce(0..terrain.long, %{}, fn(long, acc)->
      Map.put(acc, long, List.duplicate(" ", terrain.lat + 1))
    end)
  end
end